qiskit_experiments/base_experiment.py

```diff
@@ -190,7 +190,7 @@ def transpiled_circuits(self, backend=None, **kwargs):
         circuit_options = {}
         transpile_options = {}
         for key in kwargs:
-            if key in self._circuit_options:
+            if self._circuit_options is not None and key in self._circuit_options:
                 circuit_options[key] = kwargs[key]
             elif key in _TRANSPILE_OPTIONS:
                 transpile_options[key] = kwargs[key]
```

Comment:
The `kwargs` handling is about to change. Indeed we see here that it should. What if circuit options and transpile options  share the same key?

Reply:
Theoretically they could share a key with the same name but different meanings. Since they don't have to be coordinated with each other. I suggest to partition `kwargs` into separate `transpile_options, run_options, circuit_options, analysis_options`.